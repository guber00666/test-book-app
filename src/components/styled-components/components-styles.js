import styled from "styled-components";

export const ListStyles = styled.div `
              margin-top: 30px;
              border: solid 1px;
              border-radius: 0.3em;
        `;
export const PaginatorStyles = styled.div `
              margin-top: 30px;
              margin-bottom: 30px;
              float: right;
        `;
export const AppStyles = styled.div `
          margin-top: 30px;
    `;
export const CriteriaStyles = styled.div `
             margin-top: 30px;
             font-weight: bold;
    `;
export const FormStyles = styled.div`
             border: solid 1px;
             border-radius: 0.3em;
    `;
export const SearchPanelStyles = styled.div `
    border: solid 1px;
    border-radius: 0.3em;
`;