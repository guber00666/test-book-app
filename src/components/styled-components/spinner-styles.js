import styled, {keyframes} from "styled-components";

export const SpinnerPulse  = styled.div `
                width: 200px;
                height: 200px;
                display: inline-block;
                overflow: hidden;
                background: #ffffff;
                margin-left: 26em;
        `;
const SpinnerElements1 = keyframes`
    0% { top: 39.199999999999996px; height: 121.60000000000001px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements2 = keyframes`
    0% { top: 44.89999999999999px; height: 110.20000000000002px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements3 = keyframes`
    0% { top: 50.6px; height: 98.8px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
export const PulseElement = styled.div`
        width: 100%;
        height: 100%;
        position: relative;
        transform: translateZ(0) scale(1);
        backface-visibility: hidden;
        transform-origin: 0 0;
    `;
export const NthChild1 = styled.div `
         position: absolute; 
         width: 28px;
         box-sizing: content-box;
         left: 36px;
         background: #353535;
         animation: ${SpinnerElements1} 1s cubic-bezier(0,0.5,0.5,1) infinite;
         animation-delay: -0.2s
    `;
export const NthChild2 = styled.div `
          position: absolute; 
          width: 28px;
          box-sizing: content-box;
          left: 86px;
          background: #666666;
          animation: ${SpinnerElements2} 1s cubic-bezier(0,0.5,0.5,1) infinite;
          animation-delay: -0.1s
    `;
export const NthChild3 = styled.div `
           position: absolute; 
           width: 28px;
           box-sizing: content-box;
           left: 136px;
           background: #9b9b9b;
           animation: ${SpinnerElements3} 1s cubic-bezier(0,0.5,0.5,1) infinite;
           animation-delay: initial;
    `;