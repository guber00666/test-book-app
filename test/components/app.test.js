import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import CriteriaSelect from '../../src/components/criteria-select';
import FilterInput from '../../src/components/filter-input';
import List from '../../src/components/list';
import App from "../../src/components/app";
import renderer from 'react-test-renderer'
import { StyledApp } from "../../src/components/styled-components/components-styles";
import 'jest-styled-components';

configure({ adapter: new Adapter() });

describe('<App />', () => {

    const wrapperApp = shallow(
        <App/>
    );

    it('should render self and other imports', () => {
        expect(wrapperApp.find('section').hasClass("books-app")).toBe(true);
        expect(wrapperApp.find('div').hasClass("container")).toBe(true);
        expect(wrapperApp.find('h1').text()).toBe('Books');

        expect(wrapperApp.find(StyledApp)).toBeDefined();
        expect(wrapperApp.find(CriteriaSelect)).toBeDefined();
        expect(wrapperApp.find(FilterInput)).toBeDefined();
        expect(wrapperApp.find(List)).toBeDefined();

        expect(wrapperApp).toMatchSnapshot();

    });
    it('should have styles', () => {
        const styleApp = renderer.create(<StyledApp/>).toJSON();
        expect(styleApp).toHaveStyleRule('margin-top', '30px');

        expect(styleApp).toMatchSnapshot();
    })

});