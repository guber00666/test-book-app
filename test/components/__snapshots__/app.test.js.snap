// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should have styles 1`] = `
.c0 {
  margin-top: 30px;
}

<div
  className="c0"
/>
`;

exports[`<App /> should render self and other imports 1`] = `
<styled.div>
  <section
    className="books-app"
  >
    <div
      className="container"
    >
      <h1>
        Books
      </h1>
      <Connect(CriteriaSelect) />
      <Connect(FilterInput) />
      <Connect(List) />
    </div>
  </section>
</styled.div>
`;
